custom_enumerable.rb

```diff
@@ -0,0 +1,116 @@
+
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity, Style/CaseEquality
+
+
+module Enumerable
+  def my_each
+      i = 0
+      if block_given?
+      while i < self.length 
+          yield(self[i])
+          i += 1
+      end
+      end
+  end
+
+  def my_each_with_index
+    i = 0
+    for i in i...self.length
+      yield(self[i], i)
+  end
+
+  def my_select
+    array = []
+    self.my_each { |i| array << array[i] if yield(i) }
+    end
+  end  
+
+  def my_all?
+    final = true
+    self.my_each { |i| true unless yield(i) }
+    final
+  end
+
+  def my_any?
+    result = false
+    self.my_each { |i| false unless yield(i) }
+    result
+  end
+
+  def my_none?
+    result = true
+    self.my_each do |i|
+      result = false if self[i] == yield(i)
+    end
+    result
+  end
+
+  def my_count
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] counts the number of items in enum that are equal to item if an argument is given
```ruby
array.my_count(0) == array.count(0) #true
```
- [ ] counts the number of elements yielding a true value if a block is given
```ruby
array.my_count(&block) == array.count(&block)
```